src/snapshot/component_clone.rs

```diff
@@ -0,0 +1,84 @@
+use crate::{
+    GgrsComponentSnapshot, GgrsSnapshots, LoadWorld, LoadWorldSet, Rollback, RollbackFrameCount,
+    SaveWorld, SaveWorldSet,
+};
+use bevy::prelude::*;
+use std::marker::PhantomData;
+
+/// A [`Plugin`] which manages snapshots for a [`Component`] `C` using [`Clone`].
+pub struct GgrsComponentSnapshotClonePlugin<C>
+where
+    C: Component + Clone,
+{
+    _phantom: PhantomData<C>,
+}
+
+type Snapshots<C> = GgrsSnapshots<C, GgrsComponentSnapshot<C>>;
```

Comment:
I notice a bit of friction trying to understand this... perhaps I'm just tired... I also see that it's repeated a lot...

Perhaps it would make sense to move this type alias to the library crate, and expose it under a slightly more descriptive name?

As far as I understand it could just be `ComponentSnapshots<C>`?

Reply:
Yeah I think I can clarify this line a lot better. I'm popping out for a bit but when I come back I'll move these around and name them better. The basic explanation is I just needed to type out a fairly lengthy generic type 3 times, so I just made a shortcut to avoid a typo (which I had made a couple times before!)

Reply:
I've added some explicit type declarations, `GgrsResourceSnapshots` and `GgrsComponentSnapshots` (with documentation) which I believe better communicate their purpose.